packages/ai-constructs/src/conversation/runtime/types.ts

```diff
@@ -57,7 +57,17 @@ export type ConversationTurnEvent = {
       authorization: string;
     };
   };
-  messages: Array<ConversationMessage>;
+  /**
+   * @deprecated This field is going to be removed in upcoming releases.
+   */
+  messages?: Array<ConversationMessage>;
+  messageHistoryQuery: {
+    getQueryName: string;
+    getQueryInputTypeName: string;
+    listQueryName: string;
+    listQueryInputTypeName: string;
+    listQueryLimit?: number;
+  };
```

Comment:
> these values are populated by AppSync

Yes.

> are these constants or dynamic?

These are dynamic. There might be multiple "conversational routes" and associated models in AppSync. AppSync provides these names so that we do query correct route.